### PR TITLE
Add test for HandleHTTPError

### PR DESCRIPTION
**Pull request by j4qfrost**
Little contribution to get my feet wet. I want to get Github Actions setup, but from reading some of the issues it looks like there still needs to be a conversation about directory structuring around tests.

Addresses #249

### CODE REVIEW

File: backend/errors/httperror_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestHandleHTTPError(t *testing.T) {
+	tests := []struct {
+		testName       string
```
Comment (ajlacey) [inline backend/errors/httperror_test.go:15]: Please rename this to `name` to reduce repetitiveness when accessing the corresponding value.

Style reference: https://google.github.io/styleguide/go/best-practices#avoid-repetition

File: backend/errors/httperror_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	}
+	for _, test := range tests {
+		currentWriter := log.Writer()
+		var buf bytes.Buffer
+		log.SetOutput(&buf)
+		res := HandleHTTPError(test.w, test.err, test.statusCode, test.userMessage)
+		log.SetOutput(currentWriter)
+		if test.res != res {
+			t.Errorf("%s: expected %t, got %t", test.testName, test.res, res)
+		}
```
Comment (ajlacey) [inline backend/errors/httperror_test.go:53]: @j4qfrost nitpicking here, we haven't added this to our style guide yet, but in accordance with [https://google.github.io/styleguide/go/decisions#got-before-want](the google style guide for go) we prefer `got` before `want` for tests. Can you please update these before merging? The main benefit we get from this is just consistency when reading the messages in the event of a test failure/regression.

File: backend/errors/httperror_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			w:              httptest.NewRecorder(),
+			err:            errors.New("foo"),
+			statusCode:     http.StatusInternalServerError,
+			userMessage:    "bar",
+			output:         "Error: foo",
+			wrappedMessage: "{\"error\":\"bar\"}",
+			res:            true,
+		},
+	}
+	for _, test := range tests {
```
Comment (ajlacey) [inline backend/errors/httperror_test.go:45]: if you put all of the code in this loop in a `t.Run()` block [go doc](https://pkg.go.dev/testing#T.Run) you can have the test results organized by test name, which will make the output more readable for table driven tests.

File: backend/errors/httperror_test.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		currentWriter := log.Writer()
+		var buf bytes.Buffer
+		log.SetOutput(&buf)
```
Comment (ajlacey) [inline backend/errors/httperror_test.go:48]: I like how you checked the output of the log message by directing it to the buffer! :100:

**Review by ajlacey — APPROVED** [review]
@j4qfrost Thanks for taking this on, great work with these tests! If you can make the changes before merging it would be greatly appreciated, but this is solid work :)

**Comment by j4qfrost** [comment]
Cool, out and about, will have changed in in a couple hours. Thanks for looking.

File: backend/errors/httperror_test.go (hunk 3 above)
Comment (ajlacey) [reply backend/errors/httperror_test.go:45]: reference example for usage: https://go.dev/wiki/TableDrivenTests

File: backend/errors/httperror_test.go (hunk 4 above)
Comment (j4qfrost) [reply backend/errors/httperror_test.go]: Thanks, a nice trick I picked up in my Nodejs days.

**Comment by j4qfrost** [comment]
It looks like I have the 1.23 golang changes in here as well, I can rebase and remove those changes I think, or the merge can wait for that branch to be merged in first.

**Pull request merged**